comp/rdnsquerier/impl/rdnsquerier.go

```diff
@@ -181,6 +183,63 @@ func (q *rdnsQuerierImpl) GetHostname(ipAddr []byte, updateHostnameSync func(str
 	return nil
 }
 
+// GetHostnameSync attempts to resolve the hostname for the given IP address synchronously.
+// If the IP address is invalid then an error is returned.
+// If the IP address is not in the private address space then it is ignored - no lookup is performed and nil error is returned.
+// If the IP address is in the private address space then the IP address will be resolved to a hostname.
+// The function accepts a timeout duration, which defaults to 2 seconds if not provided.
+func (q *rdnsQuerierImpl) GetHostnameSync(ctx context.Context, ipAddr string) (string, error) {
+	q.logger.Infof("ENTERED THE GetHostnameSync FUNCTION with IP address: %s", ipAddr)
+
+	q.internalTelemetry.total.Inc()
```

Comment:
This function calls `GetHostname` which also increments this value, so every use of `GetHostnameSync` will be counted twice. Same for the `private` counter.

Is there any reason to duplicate the telemetry and address verification from GetHostname here?

Reply:
Good catch, thinking about this maybe it makes sense to have the public endpoints call a helper function that just contains the logic. That way the metrics can be incremented separately for each one more easily.